alpaca/rest.go

```diff
@@ -742,6 +743,155 @@ func (c *Client) GetAsset(symbol string) (*Asset, error) {
 	return &asset, nil
 }
 
+const (
+	optionContractsRequestsMaxLimit = 10000
+)
+
+func setQueryLimit(q url.Values, totalLimit, pageLimit, received, maxLimit int) {
+	limit := 0 // use server side default if unset
+	if pageLimit != 0 {
+		limit = pageLimit
+	}
+	if totalLimit != 0 {
+		remaining := totalLimit - received
+		if remaining <= 0 { // this should never happen
+			return
+		}
+		if (limit == 0 || limit > remaining) && remaining <= maxLimit {
+			limit = remaining
+		}
+	}
+
+	if limit != 0 {
+		q.Set("limit", strconv.Itoa(limit))
+	}
+}
+
+type GetOptionContractsRequest struct {
+	UnderlyingSymbols     string
+	ShowDeliverable       bool
+	Status                OptionStatus
+	ExpirationDate        civil.Date
+	ExpirationDateGTE     civil.Date
+	ExpirationDateLTE     civil.Date
+	RootSymbol            string
+	Type                  OptionType
+	Style                 OptionStyle
+	StrikePriceGTE        decimal.Decimal
+	StrikePriceLTE        decimal.Decimal
+	PennyProgramIndicator bool
+	PageLimit             int
+	TotalLimit            int
+}
+
+// GetOptionContracts returns the list of Option Contracts.
+func (c *Client) GetOptionContracts(req GetOptionContractsRequest) ([]OptionContract, error) {
+	u, err := url.Parse(fmt.Sprintf("%s/%s/options/contracts", c.opts.BaseURL, apiVersion))
+	if err != nil {
+		return nil, err
+	}
+
+	q := u.Query()
+
+	if req.UnderlyingSymbols != "" {
+		q.Set("underlying_symbols", req.UnderlyingSymbols)
+	}
+
+	q.Set("show_deliverables", strconv.FormatBool(req.ShowDeliverable))
+
+	if req.Status == OptionStatusActive || req.Status == OptionStatusInactive {
```

Comment:
[minor] I'd slightly prefer `req.Status != ""`. That way if a new status is added, the user can use it even without updating the SDK.

Reply:
Same for the type and style check.